Adding language support for url resolving

JoshuaFolkerts opened the pull request:
Added language branch option to allow for handling languages for resolving urls.

Commit:
Adding language support for url resolving

Added language branch option to allow for handling languages for resolving urls.

mrdrbob commented on ContentReferenceExtensions.cs:
```diff
@@ -157,9 +156,9 @@ public static string GetFriendlyUrl(string url, GetFriendlyUrlOption options = G
         }
 
-        public static string? ResolveUrl(this IContent content, LinkOptions options = LinkOptions.None)
+        public static string? ResolveUrl(this IContent content, string languageBranchId = "", LinkOptions options = LinkOptions.None)
```
Would it be worthwhile to also add the language branch ID to the `public static string? ResolveUrl(this ContentReference content, LinkOptions options = LinkOptions.None)` extension method below this? It would at least be a bit more consistent.

JoshuaFolkerts replied:
I agree with this change for sure.

mrdrbob commented on LinkResolverService.cs:
```diff
@@ -5,27 +5,28 @@
+        private Injected<IUrlResolver> UrlResolver { get; }
 
-        public virtual ResolvedLink? TryResolveLink(object? contentLink, LinkOptions options = LinkOptions.None)
+        public virtual ResolvedLink? TryResolveLink(object? contentLink, LinkOptions options = LinkOptions.None, string languageBranchId = "")
```
I would recommend we keep the position of the language branch ID argument consistent. Right now in some methods it's second, and others it's third. I'd suggest keeping it as the third argument, since all resolve methods take the "item to be resolved" and the options arguments, but not all of them necessarily take a language ID argument. That would keep the first two consistent across all resolve methods.

JoshuaFolkerts replied:
Agree.  I will update here.

mrdrbob commented on LinkResolverService.cs:
```diff
@@ -5,27 +5,28 @@
+                IContent content => ResolveIContent(content, languageBranchId, options),
                 Url url => ResolveUrl(url, options),
                 LinkItem linkItem => ResolveLinkItem(linkItem, options),
                 ContentReference contentReference => ResolveContentReference(contentReference, options),
```
Is there a reason we don't pass the language ID here and in the branch below? There is a method to resolve a content reference with a language identifier.

mrdrbob commented on LinkResolverService.cs:
```diff
@@ -86,13 +87,17 @@ public class LinkResolverService
             {
                 href = UrlResolver.Service.GetUrl(content);
 
+                if (!string.IsNullOrWhiteSpace(languageBranchId))
```
Can we change this so we're not potentially resolving the URL twice? Something like:
`href = string.IsNullOrWhiteSpace(languageBranchId) ? UrlResolver.Service.GetUrl(content) : UrlResolver.Service.GetUrl(content.ContentLink, languageBranchId)`?

mrdrbob commented on LinkResolverService.cs:
```diff
@@ -127,6 +134,11 @@ public class LinkResolverService
 
             string href = UrlResolver.Service.GetUrl(pageData);
 
+            if (!string.IsNullOrWhiteSpace(languageBranchId))
```
Similar to a previous comment. Can we make this something like:
`string href = string.IsNullOrWhiteSpace(languageBranchId) ? UrlResolver.Service.GetUrl(pageData) : UrlResolver.Service.GetUrl(pageData.ContentLink, languageBranchId);`?

mrdrbob reviewed (changes requested):
Overall I think is a good and necessary change. I just had a few questions / requests. Let me know if I'm off base anywhere here. Thanks!

Commit:
Making changes to method for more consistant approach

Moving properties in method as last argument for more consistancy.  Also adding methods for more consisistancy for extension methods
